### PR TITLE
[selinux] More precise warnings when policy install fails

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Make failure instructions more detailed.
Specifies that the SELinux steps are only useful if using system-probe.

### Motivation

More precise install logs.

### CODE REVIEW

File: omnibus/package-scripts/agent/postinst [hunk 1]
```diff
@@ -91,23 +91,37 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
...
+            echo "Loading SELinux policy module for system-probe."
             semodule -v -i $CONFIG_DIR/selinux/system_probe_policy.pp >/dev/null 2>&1
             if [ "$?" != "0" ]; then
-                echo "The SELinux policy module for datadog-agent is not compatible with this host."
-                echo "Please install or update the selinux-policy-default package and then reinstall the datadog-agent package to configure SELinux for the Datadog Agent."
-                echo "Skipping SELinux configuration."
+                echo "Couldn’t load system-probe policy."
+                echo "To be able to run system-probe on your host, please install or update the"
+                echo "selinux-policy-default and policycoreutils-python-utils packages."
+                echo "Then run the following commands:"
```
Comment (albertvaka) [inline omnibus/package-scripts/agent/postinst]:
```suggestion
                echo "Then run the following commands, or reinstall datadog-agent:"
```
Wdyt?

File: omnibus/package-scripts/agent/postinst [hunk 2]
```diff
@@ -91,23 +91,37 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
...
+                        echo "    restorecon -v $INSTALL_DIR/embedded/bin/system-probe"
                     fi
                 else
-                    echo "Could not set up SELinux labels for datadog-agent: semanage and/or restorecon commands not found."
-                    echo "Please make sure they are installed (via the policycoreutils, policycoreutils-python and/or policycoreutils-python-utils packages), and then run:"
+                    echo "Couldn’t load system-probe policy (missing selinux utilities)"
+                    echo "To be able to run system-probe on your host, please install or update the"
+                    echo "selinux-policy-default and policycoreutils-python-utils packages."
+                    echo "Then run the following commands:"
+                    echo "    semodule -i $CONFIG_DIR/selinux/system_probe_policy.pp"
```
Comment (albertvaka) [inline omnibus/package-scripts/agent/postinst]: I guess this line is not needed? It wasn't there before. Unless you want to display the same instructions for all error cases.

File: omnibus/package-scripts/agent/postinst [hunk 3]
```diff
@@ -91,23 +91,37 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
...
-            echo "Loading SELinux policy module for datadog-agent."
+            echo "Loading SELinux policy module for system-probe."
             semodule -v -i $CONFIG_DIR/selinux/system_probe_policy.pp >/dev/null 2>&1
             if [ "$?" != "0" ]; then
-                echo "The SELinux policy module for datadog-agent is not compatible with this host."
-                echo "Please install or update the selinux-policy-default package and then reinstall the datadog-agent package to configure SELinux for the Datadog Agent."
-                echo "Skipping SELinux configuration."
+                echo "Couldn’t load system-probe policy."
+                echo "To be able to run system-probe on your host, please install or update the"
+                echo "selinux-policy-default and policycoreutils-python-utils packages."
```
Comment (albertvaka) [inline omnibus/package-scripts/agent/postinst]:
```suggestion
                echo "selinux-policy-default and policycoreutils-python (or policycoreutils-python-utils depending on your distribution) packages."
```
Shouldn't we mention both here as well?

File: omnibus/package-scripts/agent/postinst [hunk 4]
```diff
@@ -91,23 +91,37 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
...
+                echo "Then run the following commands:"
+                echo "    semodule -i $CONFIG_DIR/selinux/system_probe_policy.pp"
+                echo "    semanage fcontext -a -t system_probe_t $INSTALL_DIR/embedded/bin/system-probe"
+                echo "    restorecon -v $INSTALL_DIR/embedded/bin/system-probe"
             else
                 echo "Labeling SELinux type for the system-probe binary."
                 if command -v semanage >/dev/null 2>&1 && command -v restorecon >/dev/null 2>&1;then
                     semanage fcontext -a -t system_probe_t $INSTALL_DIR/embedded/bin/system-probe && restorecon -v $INSTALL_DIR/embedded/bin/system-probe
                     if [ "$?" != "0" ]; then
-                        echo "Could not relabel the system-probe binary with type system_probe_t."
```
Comment (albertvaka) [inline omnibus/package-scripts/agent/postinst]: We didn't have instructions here, now we do. Why the change? I guess it might be worth creating a function that prints these, since we have essentially the same instructions three times.

File: omnibus/package-scripts/agent/postinst (hunk 2 above)
Comment (KSerrania) [reply omnibus/package-scripts/agent/postinst]: In the future (once this is refactored), all cases will show the same lines, so I think we can start doing this right now. Loading the module twice is not a problem (modules are named and versioned, so the loading the same module twice is idempotent).

File: omnibus/package-scripts/agent/postinst (hunk 1 above)
Comment (KSerrania) [reply omnibus/package-scripts/agent/postinst]: Yeah, we can add this

File: omnibus/package-scripts/agent/postinst (hunk 3 above)
Comment (albertvaka) [reply omnibus/package-scripts/agent/postinst]: No need then. Sorry, it wasn't obvious to me that this script is only used on Debian.

File: omnibus/package-scripts/agent/postinst (hunk 4 above)
Comment (albertvaka) [reply omnibus/package-scripts/agent/postinst]: Ok, we can do this for 7.19 then.

**Review by albertvaka — DISMISSED** [review]
LGTM. I would add "or reinstall datadog-agent" in the instructions since that seems a simpler way to achieve the same to me. Ping me if you add it so I approve again.

**Review by albertvaka — APPROVED** [review]
